optimizer/plan/planbuilder_join.go

```diff
@@ -209,7 +209,7 @@ func (p *joinPath) attachCondition(condition ast.ExprNode, availablePaths []*joi
 		p.filterRate *= filterRate
 		return true
 	}
-	if p.inner.attachCondition(condition, append(availablePaths, p.outer)) {
+	if p.inner.attachCondition(condition, availablePaths) {
```

Comment:
If this is an ON condition, we should push it down to inner with `p.outer` as availablePath.